zebedee-reader/src/test/java/com/github/onsdigital/zebedee/util/URIUtilsTest.java

```diff
@@ -0,0 +1,215 @@
+
+import static org.junit.Assert.*;
+
+public class URIUtilsTest {
+
+    @Test
+    public void removeLastSegmentWhenEndingOnEmpty(){
+        String result = URIUtils.removeLastSegment("//");
+        assertEquals("/",result);
+    }
+
+    @Test
+    public void removeLastSegmentWhenEndingOnSlash(){
+        String result = URIUtils.removeLastSegment("/segment1/segment2/segment3/");
+        assertEquals("/segment1/segment2",result);
+    }
+
+    @Test
+    public void removeLastSegment(){
+        String result = URIUtils.removeLastSegment("/segment1/segment2/segment3");
+        assertEquals("/segment1/segment2",result);
+    }
+
+    @Test
+    public void getDecodedUrlReturnsDecodedUrlString() {
+        String result = URIUtils.getDecodedUrl("%2fsegment1%2fsegment2%2fsegment3%2f...");
+        assertEquals("/segment1/segment2/segment3/...",result);
+    }
+
+    @Test
+    public void isURLEncodedReturnTrue(){
+        assertTrue(URIUtils.isUUEncoded("%2fsegment1%2fsegment2%2fsegment3%2f..."));
+    }
+
+    @Test
+    public void isURLEncodedReturnFalse() {
+        assertFalse(URIUtils.isUUEncoded("/segment1/segment2/segment3/..."));
+    }
+
+    @Test
+    public void isURLEncodedReturnFalseForSlash() {
+        assertFalse(URIUtils.isUUEncoded("/"));
+    }
+
+    @Test
+    public void getQueryParameterFromURLTest() {
+        String url = "/file?URI=" + "/Economy/inflation/articles/research/topic/health/december2023/december2023/";
+        String result = URIUtils.getQueryParameterFromURL(url, "Uri");
+
+        assertEquals("/economy/inflation/articles/research/topic/health/december2023/december2023/", result);
+    }
+
+    @Test
+    public void getParameterMapFromQueryTest() {
+        Map<String, String> map = new HashMap<>();
+        map.put("uri", "/economy/inflation/articles/research/topic/health/december2023/december2023/");
+
+        String url = "URI=" + "/Economy/inflation/articles/research/topic/health/december2023/december2023/";
+        Map<String, String> result = URIUtils.getParameterMapFromQuery(url);
+
+        assertEquals(map.get("uri"), result.get("uri"));
+    }
+
+    @Test
+    public void removeAfterKeywordTest() {
+        String url = "/economy/inflation/articles/research/topic/health/december2023/december2023.json";
+        String result = URIUtils.removeAfterKeyword(url, "/articles");
+        assertEquals("/economy/inflation", result);
+    }
+
+    @Test
+    public void removeAfterKeywordWhenKeywordNotFoundReturnSameTest() {
+        String url = "/economy/inflation/articles/research/topic/health/december2023/december2023.json";
+        String result = URIUtils.removeAfterKeyword(url, "/bulletin");
+        assertEquals("/economy/inflation/articles/research/topic/health/december2023/december2023.json", result);
+    }
+
+    @Test
+    public void isLastSegmentFileExtensionYes() {
+        String url = "/economy/inflation/articles/research/topic/health/december2023/december2023.json";
```

Comment:
Is it worth adding more tests here with URLs containing different file extensions?